tests/Microsoft.Identity.Test.Unit/CacheTests/CachePartitioningTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.CacheTests
+{
+    [TestClass]
+    public class CachePartitioningTests : TestBase
+    {
+        [TestMethod]
+        public async Task CCAFlows_CachePartition_TestAsync()
+        {
+            using (var harness = base.CreateTestHarness())
+            {
+                var httpManager = harness.HttpManager;
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                              .WithRedirectUri(TestConstants.RedirectUri)
+                                                              .WithClientSecret(TestConstants.ClientSecret)
+                                                              .WithHttpManager(httpManager)
+                                                              // this will fail if cache partitioning rules are broken (but not when cache serialization is also used)
+                                                              .WithCachePartitioningAsserts(harness.ServiceBundle.PlatformProxy)
+                                                              .BuildConcrete();
+
+                await RunClientCreds_Async(httpManager, app).ConfigureAwait(false);
+
+                await RunObo_Async(httpManager, app).ConfigureAwait(false);
+
+                await RunAuthCode_Async(httpManager, app).ConfigureAwait(false);
+            }
+        }
+
+        [TestMethod]
+        public async Task PCAFlows_CachePartition_TestAsync()
```

Comment:
@bgavrilMS This seems duplicate, was it supposed to be PCA code?

Reply:
Yeah, I think I modified some existing PCA tests instead.